og.module

```diff
@@ -164,6 +164,20 @@ function og_entity_access(EntityInterface $entity, $operation, AccountInterface
   /** @var \Drupal\Core\Access\AccessResult $access */
   $access = \Drupal::service('og.access')->userAccessEntityOperation($operation, $entity, $account);
 
+  $audience_fields = \Drupal::service('og.group_audience_helper')->getAllGroupAudienceFields($entity_type_id, $bundle_id);
+  if (count($audience_fields) === 1) {
+    $field_definitions = \Drupal::service('entity_field.manager')->getFieldDefinitions($entity_type_id, $bundle_id);
+    foreach ($field_definitions as $field_definition) {
+      /** @var \Drupal\Core\Field\FieldDefinitionInterface $field_definition */
+      if (!in_array($field_definition->getName(), array_keys($audience_fields))) {
+        continue;
+      }
+
+      // Check if the field needs to be required.
+      $field_definition->setRequired($access);
+    }
+  }
+
```

Comment:
This is logic is undocumented, untested and seemingly unneeded. Altering field definitions to set them required inside `hook_entity_access()`? This is completely bizarre and can not be the correct solution for whatever this is trying to fix.

This section MUST be removed.